test/test_app.py

```diff
@@ -58,3 +58,22 @@ def test_carriage_return_sql(client):
 
     # we should get an extra z in there if the carriage returns are not well handled.
     assert fixed_sql.count("z") == 1
+
+
+def test_unrecoverable_failure_sql(client):
+    """Test unrecoverable failure
+
+    If we pass templates the online tester can't cope (See issue #25).
+    Make sure it fails gracefully.
+    """
+    sql_encoded = sql_encode(
+        "INSERT into {{ params.schema }}.daily_service_metrics (\nSELECT\n*\nFROM table\n)"
+    )
+    rv = client.get("/fluffed", query_string=f"""dialect=ansi&sql={sql_encoded}""")
+
+    html = rv.data.decode().lower()
+    soup = BeautifulSoup(html, "html.parser")
+    fixed_sql = soup.find("textarea", {"id": "fixedsql"}).text
+
+    # we should get an extra z in there if the carriage returns are not well handled.
```

Comment:
It looks like this code was copied from the test above, which tests a query whose final character is a `z`. I don't _think_ it's needed in this test.

Reply:
Even more lazy of me! Proper test included now.